Plugin/Model/Convert/OrderPlugin.php

```diff
@@ -74,17 +78,16 @@ public function afterToInvoice(\Magento\Sales\Model\Convert\Order $subject, $res
             $orderExtensionAttributes = $this->orderExtensionFactory->create();
         }
         if ($orderExtensionAttributes->getShippingProtection() === null) {
+
             $shippingProtectionTotalData = $this->shippingProtectionTotalRepository->get($order->getEntityId(), ShippingProtectionTotalInterface::ORDER_ENTITY_TYPE_ID);
             if($shippingProtectionTotalData->getData()) {
-                $orderExtensionAttributes->setShippingProtection(
-                    [
-                        'base' => $shippingProtectionTotalData->getShippingProtectionBasePrice(),
-                        'base_currency' => $shippingProtectionTotalData->getShippingProtectionBaseCurrency(),
-                        'price' => $shippingProtectionTotalData->getShippingProtectionPrice(),
-                        'currency' => $shippingProtectionTotalData->getShippingProtectionCurrency(),
-                        'sp_quote_id' => $shippingProtectionTotalData->getSpQuoteId()
-                    ]
-                );
+                $shippingProtection = $this->shippingProtectionFactory->create();
```

Comment:
Can we use `getAndSaturateExtensionAttributes` here?

Reply:
I was hesitant to do this here because this specific method first checks to make sure the extension attributes property exists, and if it doesn't it creates it.  That's incompatible with the current version of that function.